### PR TITLE
copr: Enable automatic RPM build on every PR merge

### PR DESCRIPTION
**Pull request by lsm5**
This commit along with a github webhook will automatically trigger rpm builds on the `rhcontainerbot/podman-next` copr after every PR merge.

Useful for those who can't wait for releases and are ok with danger.

Ref: https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

### What does this PR do?

will automatically trigger rpm builds on `rhcontainerbot/podman-next` copr.

### Screenshot/screencast of this PR

Not relevant. But here's the link to the builds triggered on the copr: https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/package/podman-desktop/

### What issues does this PR fix or reference?

None. Though loosely related to https://github.com/containers/podman-desktop/issues/112 

### How to test this PR?

As long as every PR successfully triggers an RPM build on the copr, we're good.

/cc @rhatdan

### CODE REVIEW

**Review by benoitf — CHANGES_REQUESTED** [review]
we already have a https://github.com/containers/podman-desktop/blob/main/.flatpak.desktop

if it's for rpm, should be prefixed by rpm or move to a .rpm directory along with the other resources

File: podman-desktop.spec.rpkg
Comment (benoitf) [inline podman-desktop.spec.rpkg:2]: I don't think it should belong in this repository.

Why do we need this rpkg file over the rpm that can be produced by electron-builder ?

This file probably don't need to be there
For example flathub (that is generating flatpak files contains all the information it its own repository https://github.com/flathub/io.podman_desktop.PodmanDesktop

Comment (lsm5) [reply podman-desktop.spec.rpkg:2]: 1. I have a separate [spec file](https://raw.githubusercontent.com/lsm5/podman-desktop-rpm/main/podman-desktop.spec) at [podman-desktop-rpm](https://github.com/lsm5/podman-desktop-rpm) which will generate a new RPM from every upstream release. Those rpms will get posted to a separate copr: [rhcontainerbot/podman-desktop](https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-desktop/builds/). That spec file is still far from being the official Fedora spec file but is a step in that direction. The rpm generated by electron-builder in my observation used FPM which can't be used for any official Fedora or RHEL inclusion. Nor would flatpak packages help towards that. So, we need rpm spec files if we're looking at any sort of official downstream inclusion.

2. This rpkg file serves a different purpose than the spec file mentioned in [1]. This rpkg spec file will build RPMs from the latest upstream main branch so that users looking for the latest bits can try them out quickly and conveniently. This is already being done upstream for [Podman](https://github.com/containers/podman/blob/main/podman.spec.rpkg), [Buildah](https://github.com/containers/buildah/blob/main/buildah.spec.rpkg), [Skopeo](https://github.com/containers/skopeo/blob/main/skopeo.spec.rpkg), [Netavark](https://github.com/containers/netavark/blob/main/netavark.spec.rpkg), and [Aardvark-dns](https://github.com/containers/aardvark-dns/blob/main/aardvark-dns.spec.rpkg). All those RPMs are available at a [single COPR repo](https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/packages/), so it's a 1-stop shop for people looking to use the very latest container tools. There's at least 1 RH customer using the `rhcontainerbot/podman-next` COPR for early product integration. Also, see a relevant [blog post](https://hobo.house/2017/09/03/automate-rpm-builds-from-git-sources-using-copr/).

3. Electron-builder will need every user to build the rpm themselves every time they need to update to the latest. This rpkg file takes that hassle away. Users will only need to enable 1 single COPR and then just  `dnf update` to get the latest of everything.

4. This rpkg file comes with effectively 0 additional cost to upstream. It does not affect the upstream PR workflow, and works on the fedora COPR infra only after the PR is merged. Also, with this PR, I'm effectively signing up to maintain that file and the COPR builds myself. So, it shouldn't be any extra burden on the current upstream maintainers.

Comment (benoitf) [reply podman-desktop.spec.rpkg:2]: so you convinced me that it should not be part of this repository as it's not buildable outside of a dedicated environment.

Comment (lsm5) [reply podman-desktop.spec.rpkg:2]: > so you convinced me that it should not be part of this repository as it's not buildable outside of a dedicated environment.

well, it's buildable using `rpkg local` on anyone's fedora installation or any distro that ships rpkg for that matter. So, a dedicated environment is not needed for that.

**Comment by lsm5** [comment]
@benoitf addressed the desktop file issue. Please let me know if you have any other unanswered concerns or if you're still opposed regardless.

**Comment by benoitf** [comment]
only the package.json change can be merged.

The other parts are specific to the RPM consumer.

it's like for flathub repository [1] . It depends on the source and add the specific flathub metadata on its own repository.


In this podman-desktop repository, there is nothing about RPM. If it's outside the scope of nodejs/electron/electron-builder then it's not in the repository.

[1] https://github.com/flathub/io.podman_desktop.PodmanDesktop

**Comment by lsm5** [comment]
alright, thanks.

**Pull request closed without merging**
